src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToTestAMultitargetedSolutionWithPublishReleaseOrPackRelease.cs

```diff
@@ -0,0 +1,395 @@
+        /// <param name="PReleaseProperty">The value of the property to set, PublishRelease or PackRelease in this case.</param>
+        /// <param name="exePReleaseValue">If "", the property will not be added. This does not undefine the property.</param>
+        /// <param name="libraryPReleaseValue">If "", the property will not be added. This does not undefine the property.</param>
+        /// <param name="testPath">Use to set a unique folder name for the test, like other test infrastructure code.</param>
+        /// <returns></returns>
+        internal Tuple<TestSolution, List<TestProject>> Setup(ITestOutputHelper log, List<string> exeProjTfms, List<string> libraryProjTfms, string PReleaseProperty, string exePReleaseValue, string libraryPReleaseValue, [CallerMemberName] string testPath = "")
+        {
+            // Project Setup
+            List<TestProject> testProjects = new List<TestProject>();
+            var testProject = new TestProject("TestProject")
+            {
+                TargetFrameworks = String.Join(";", exeProjTfms),
+                IsExe = true
+            };
+            testProject.RecordProperties("Configuration", "Optimize", PReleaseProperty);
+            if (exePReleaseValue != "")
+            {
+                testProject.AdditionalProperties[PReleaseProperty] = exePReleaseValue;
+            }
+            var mainProject = _testAssetsManager.CreateTestProject(testProject, callingMethod: testPath, identifier: string.Join("", exeProjTfms) + PReleaseProperty);
+
+            var referencedProject = new TestProject("ReferencedProject")
+            {
+                TargetFrameworks = String.Join(";", libraryProjTfms),
+                IsExe = false
+            };
+            referencedProject.RecordProperties("Configuration", "Optimize", PReleaseProperty);
+            if (libraryPReleaseValue != "")
+            {
+                referencedProject.AdditionalProperties[PReleaseProperty] = libraryPReleaseValue;
+            }
+            var secondProject = _testAssetsManager.CreateTestProject(referencedProject, callingMethod: testPath, identifier: string.Join("", libraryProjTfms) + PReleaseProperty);
+
+            List<TestAsset> projects = new List<TestAsset> { mainProject, secondProject };
+
+            // Solution Setup
+            var sln = new TestSolution(log, mainProject.TestRoot, projects);
+            testProjects.Add(testProject);
+            testProjects.Add(referencedProject);
+            return new Tuple<TestSolution, List<TestProject>>(sln, testProjects);
+        }
+
+
+
+        [InlineData(PublishRelease, "net8.0", true)]
+        [InlineData(PublishRelease, "-p:TargetFramework=net8.0", false)]
+        [Theory]
+        public void ItUsesReleaseWithATargetFrameworkOptionNet8ForNet6AndNet7MultitargetingProjectWithPReleaseUndefined(string releaseProperty, string args, bool passDashF)
```

Comment:
Also perhaps the parameters should change to use `ToolsetInfo.CurrentTargetFramework`, so that when that revs you're not passing `-f net8.0` for a project that doesn't normally target `net8.0`.

Reply:
Thanks, this used to be a pack test too but turned into just publish